notebooks/403-action-recognition-webcam/403-action-recognition-webcam.ipynb

```diff
@@ -0,0 +1,646 @@
+    "    input_key = list(exec_net.input_info)[0]\n",
+    "    output_keys = list(exec_net.outputs.keys())\n",
+    "    return input_key, output_keys, exec_net"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Initialization for Encoder and Decoder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Encoder initialization \n",
+    "input_key_en, output_keys_en, exec_net_en = model_init(model_path_encoder, model_weights_path_encoder)\n",
+    "#Decoder initialization \n",
+    "input_key_de, output_keys_de, exec_net_de = model_init(model_path_decoder, model_weights_path_decoder)\n",
+    "\n",
+    "# get input size - Encoder\n",
+    "height_en, width_en = exec_net_en.input_info[input_key_en].tensor_desc.dims[2:]\n",
+    "# get input size - Decoder\n",
+    "height_de = exec_net_de.input_info[input_key_de].tensor_desc.dims[2:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Helper functions\n",
+    "\n",
+    "We will use the next AI functions to run the architecture Encoder-Decoder of ResNet34\n",
+    "\n",
+    "1. Preprocess the input image before running the Encoder model. (`center_crop` and `adaptative_resize`)\n",
+    "2. Decode top-3 probabilities into label names. (`decode_output`)\n",
+    "\n",
+    "Also, we can see here the minimal configuration to display text over the live streaming or video for the main process.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def center_crop(frame: np.array) -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Center crop squared the original frame to standardize the input image to the encoder model\n",
+    "    \n",
+    "    :param frame: input frame\n",
+    "    :returns: center-crop-squared frame\n",
+    "    \"\"\"\n",
+    "    img_h, img_w, _ = frame.shape\n",
+    "    min_dim = min(img_h, img_w)\n",
+    "    start_x = int((img_w - min_dim) /2.)\n",
+    "    start_y = int((img_h - min_dim) /2.)\n",
+    "    return frame[start_y:start_y+min_dim, start_x:start_x+min_dim, ...]\n",
+    "\n",
+    "def adaptive_resize(frame: np.array, size: int) -> np.ndarray:\n",
+    "    \"\"\"\n",
+    "    Adaptative resize of the cropped image to adjust to the input size model 224\n",
+    "    \n",
+    "    :param frame: input frame\n",
+    "    :param size: input size to encoder model\n",
+    "    :returns: resized frame, np.array type \n",
+    "    \"\"\"\n",
+    "    h, w, _ = frame.shape\n",
+    "    scale = size / min(h, w)\n",
+    "    ow, oh = int(w * scale), int(h * scale)\n",
+    "    if ow == w and oh == h:\n",
+    "        return frame\n",
+    "    return cv2.resize(frame, (ow, oh))\n",
+    "\n",
+    "def decode_output(probs: np.array, labels: np.array, top_k: int = None) -> np.ndarray:\n",
```

Comment:
`def decode_output(probs: np.ndarray, labels: np.ndarray, top_k: Optional[int] = None) -> np.ndarray:`

And then add `from typing import Optional` to the imports. (But see next comment, I suggest making this non optional with 3 as default).